### PR TITLE
Make sure the dst dir doesn't exist when calling shutil.copytree

### PR DESCRIPTION
**Pull request by jjyao**
Signed-off-by: Jiajun Yao <jeromeyjj@gmail.com>

## Why are these changes needed?
In https://github.com/ray-project/ray/pull/29603, we replaced `distutils.dir_util.copy_tree` with `shutil.copytree` since the former is deprecated. There is a big difference between these two methods that we ignored and caused test failures: `distutils.dir_util.copy_tree` works regardless whether dst dir exists or not but `shutil.copytree` expects that the dst dir doesn't exist.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
This is one example of test failure: https://buildkite.com/ray-project/oss-ci-build-branch/builds/881#018436fc-cde7-4c1f-88f1-2b59bbe984a5.

The reason why it's not caught in the original PR is that the test was not triggered based on the logic of deciding which tests to run.

**Review by rickyyx — APPROVED** [review]
<div><br/>via <a href="https://giphy.com/gifs/5nsiFjdgylfK3csZ5T">GIPHY</a></div>


Oh nice!

File: .buildkite/pipeline.build.yml
Comment (jjyao) [inline .buildkite/pipeline.build.yml:173]: This is just to trigger the test, will revert.

Comment (rickyyx) [reply .buildkite/pipeline.build.yml:173]: I think if you do `[all_test]` in the original PRs, you will get that also.

Comment (jjyao) [reply .buildkite/pipeline.build.yml:173]: Learned something new!!

**Comment by jjyao** [comment]
`Ray CPP Worker` passed: https://buildkite.com/ray-project/oss-ci-build-pr/builds/3984#018439ba-9875-42ad-875a-f8e39e073e7b

File: python/ray/scripts/scripts.py
Comment (SongGuyang) [inline python/ray/scripts/scripts.py:2513]: So currently, users don't need to provide the dir? Can you also update the doc https://github.com/ray-project/ray/blame/master/doc/source/ray-overview/installation.rst#L178? (remove the command `mkdir ray-template`)

**Review by SongGuyang — APPROVED** [review]
LGTM if you can update the doc before merge.

**Review by maxpumperla — APPROVED** [review]

**Pull request merged**
